### PR TITLE
Feature/basic infrastructure

### PR DESCRIPTION
**Pull request by crnopster**
Initialize modules, gitignore, dockerfile, docker-compose file.

Docker compose contains:

graylog
consul
kafka
postges
service image

### CODE REVIEW

**Review by alpaka-alpachino — COMMENTED** [review]
Greate 👍  You may just add discription

**Review by maksymbilokur — DISMISSED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+WORKDIR /opt/resource/
+COPY cmd             cmd
+COPY config          config
+COPY kafka           kafka
+COPY filter          filter
+COPY matcher         matcher
+COPY storage         storage
+COPY web             web
+#COPY go.*            ./
```
Comment (dmitriyminer) [inline Dockerfile]: Is it dead code? Let's remove it

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+WORKDIR /opt/resource/cmd/
+RUN go build -o /opt/services/example-app .
+
+FROM alpine:3.7
+COPY --from=builder /opt/services/example-app /opt/services/example-app
```
Comment (dmitriyminer) [inline Dockerfile]: Let's rename to project app name

File: Dockerfile (hunk 1 above)
Comment (crnopster) [reply Dockerfile]: its for future development. We can add some code to any of this folders and run docker-compose without editing Dockerfile

File: config/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,5 @@
+FROM consul:1.7.1
+
```
Comment (dmitriyminer) [inline config/Dockerfile]: WDYT about structure your infra configurations and Dockerfiles, for example:
```
--infra
    -- consul
        agent-config.json
        Dockerfile
    -- ...
```

File: Dockerfile (hunk 1 above)
Comment (dmitriyminer) [reply Dockerfile]: `#COPY go.*            ./` ?

File: Dockerfile (hunk 2 above)
Comment (crnopster) [reply Dockerfile]: renamed to positioning-filter

File: config/Dockerfile (hunk 3 above)
Comment (crnopster) [reply config/Dockerfile]: moved to infra/consul/

File: Dockerfile (hunk 1 above)
Comment (crnopster) [reply Dockerfile]: removed

**Review by dmitriyminer — APPROVED** [review]

**Review by alpaka-alpachino — APPROVED** [review]

**Review by VasylKosiv — APPROVED** [review]

**Review by maksymbilokur — APPROVED** [review]

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    container_name: elasticsearch
+    environment:
+      - http.host=0.0.0.0
+      - transport.host=localhost
+      - network.host=0.0.0.0
+      - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
+    ulimits:
+      memlock:
+        soft: -1
+        hard: -1
```
Comment (radikh) [inline docker-compose.yaml:42]: Why that parameters are here?

File: docker-compose.yaml [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    container_name: mongodb
+    
+  elasticsearch:
+    image: docker.elastic.co/elasticsearch/elasticsearch-oss:6.8.2
+    container_name: elasticsearch
+    environment:
+      - http.host=0.0.0.0
+      - transport.host=localhost
+      - network.host=0.0.0.0
+      - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
```
Comment (radikh) [inline docker-compose.yaml:38]: What do these parameters do?

File: README.md [hunk 6]
```diff
@@ -52,19 +51,20 @@ Filter type can be rectangular or round(in future maybe shaped)
...
-MIGRATIONS_PATH -path to folder with .sql migrations files  
+PF_USER -default user for accesing database from api  
+PF_PASSWORD -password for user
 
-### commands
+### Commands  
 
-docker run -v "$MIGRATIONS_PATH:/migrations"  --network host migrate/migrate -path=/migrations/ -database "postgresql://$POSTGRES_USER:$POSTGRES_PASSWORD@localhost:$DB_POST/$POSTGRES_DB?sslmode=disable" up
+```bash
+docker-compose up --build -d
```
Comment (radikh) [inline README.md:69]: How to ensure that all is set up correctly?

File: docker-compose.yaml [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+    ports:
+      # HTTP
+      - 9000:9000
+      # Syslog TCP
+      - 1514:1514
+      # Syslog UDP
+      - 1514:1514/udp
+      # GELF TCP
+      - 12201:12201
+      # GELF UDP
```
Comment (radikh) [inline docker-compose.yaml:65]: I'm not really sure that I understand all of these comments. Could you please  describe them to me?

File: go.mod [hunk 8]
```diff
@@ -1,3 +1,5 @@
 module github.com/lvl484/positioning-filter
 
 go 1.13
+
+require github.com/google/uuid v1.1.1
```
Comment (radikh) [inline go.mod:5]: Why is it required?

File: Dockerfile [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+COPY --from=modules /go/pkg/ /go/pkg/
+
+WORKDIR /opt/resource/
+COPY cmd             cmd
+COPY config          config
+COPY kafka           kafka
+COPY filter          filter
+COPY matcher         matcher
+COPY storage         storage
+COPY web             web
```
Comment (radikh) [inline Dockerfile:17]: Is it reasonable to add each package to the docker build?

File: infra/consul/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,5 @@
+FROM consul:1.7.1
+
+COPY agent-config.json /consul/configs/
+
+ENTRYPOINT ["consul", "agent", "-config-dir=/consul/configs", "-data-dir=/tmp/consul"]
```
Comment (radikh) [inline infra/consul/Dockerfile:5]: Is there any way to do this stuff in docker compose file?

Comment (crnopster) [reply infra/consul/Dockerfile:5]: I did not find way to do that in docker compose file. I had a problem with docker compose to create config for consul agent. Dockerfile was the best solution for me

File: go.mod (hunk 8 above)
Comment (crnopster) [reply go.mod:5]: check c41996b

File: docker-compose.yaml (hunk 7 above)
Comment (crnopster) [reply docker-compose.yaml:65]: Syslog - "Graylog is able to accept and parse RFC 5424 and RFC 3164 compliant syslog messages and supports TCP and UDP." - graylog docs.
GELF - Graylog Extended Log Format is used for positioning-filter service
HTTP - graylog web interface

**Pull request merged**